File: Sorting Algorithms/Swift/bubbleSort.swift
func swap(Arr: inout [Int], x: Int, y: Int) {
    let temp = Arr[x]
    Arr[x] = Arr[y]
    Arr[y] = temp
}

func bubbleSort(Arr: inout [Int]) {
    for _ in 0 ..< Arr.count {
        for i in 0 ..< Arr.count - 1 {
            if Arr[i] > Arr[i+1] {
                swap(Arr: &Arr, x: i, y: i+1)
            }
        }
    }
}

let unsortedIntegers = [5, 8, 3, 4, -1, 3, 8, 32423423]
var sortedIntegers = unsortedIntegers
bubbleSort(Arr: &sortedIntegers)

print("Unsorted Integers: \(unsortedIntegers)")
print("Sorted Integers: \(sortedIntegers)")

File: Sorting Algorithms/Rust/bubble_sort.rs

fn main() {
    let mut array = vec![10, 4, 8, 4, 9, 4, 0];
    println!("Using vector: {:?}", array);

    bubble_sort(array.as_mut());
    println!("Sorted: {:?}", array);

}


fn bubble_sort(vector: &mut Vec<i32>) {
    let vector_size = vector.len();
    for _ in 0..vector_size {
        for i in 1..vector_size {
            if vector[i] < vector[i-1] {
                let temp = vector[i];
                vector[i] = vector[i-1];
                vector[i-1] = temp;
            }
        }
    }
}